src/analysis/Environments/Environments.models.ts

```diff
@@ -22,5 +22,9 @@ export interface LeoResourcePermissionsProvider {
   canPauseResource: (resource: App | ListRuntimeItem) => boolean;
 }
 
+export interface CanDeleteProvider {
+  canBeDeleted: (resource: App) => boolean;
```

Comment:
this can live in the `LeoResourcePermissionProvider` and shouldn't need its own interface.